### ISSUE TITLE
Certificate errors in distributed mode with docker compose

### ISSUE THREAD

**Issue by MikeLReminderM** [minio/minio#10490]
I have a three node distributed mode cluster that has been working fine over plain http.   I need to make it run over https.   I've not been able to get past certificate and TLS handshake errors.

## Expected Behavior

The expected behavior would be for the cluster to come up with no errors and be reachable over a https URL instead of plain http.

## Current Behavior

I experience the following two errors now so it seems the instances have issues with each other's certs:

minio2_1            | Unable to read 'format.json' from https://minio3.local:9000/data4: Post https://minio3.local:9000/minio/storage/data4/v20/readall?disk-id=&file-path=format.json&volume=.minio.sys: x509: certificate is not valid for any names, but wanted to match minio3.local

minio2_1            | API: SYSTEM()
minio2_1            | Time: 14:41:15 UTC 09/15/2020
minio2_1            | Error: http: TLS handshake error from 172.18.0.4:49384: remote error: tls: bad certificate
minio2_1            |        1: github.com/minio/minio@/cmd/server-main.go:470:cmd.serverMain.func2()

## Steps to Reproduce (for bugs)

1.  create a private key and wildcard cert for *.local (the instances are named mini[1-3].local):

openssl genrsa -aes256 -out private.key 2048 -passout pass:secret

openssl req -new -x509 -days 3650 -key private.key -out public.crt -subj "/C=US/ST=PA/L=KOP/O=Company/CN=<*.local>"

2. make a /tmp/.minio folder on the host containing a certs folder and CAs folder within certs
    copy both the private.key and public.crt in to "certs", and copy public.crt in to "CAs"

3. modify the docker-compose.yml 

OLD plain http minio section for minio1 as an example from the docker-compose.yml:

  minio1:
    image: minio/minio:RELEASE.2020-08-04T23-10-51Z
    volumes:
      - minio1-1:/data1
      - minio1-2:/data2
      - minio1-3:/data3
      - minio1-4:/data4
    ports:
      - "9003:9000"
    environment:
      MINIO_ACCESS_KEY: minio
      MINIO_SECRET_KEY: secret
    command: server http://minio{1...3}/data{1...4}
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

NEW minio section to change to https for minio1 for an example from the docker-compose.yml:

  minio1:
    image: minio/minio:RELEASE.2020-08-04T23-10-51Z
    volumes:
      - minio1-1:/data1
      - minio1-2:/data2
      - minio1-3:/data3
      - minio1-4:/data4
# NEW volume mount for minio to see the cert and key
      - /tmp/.minio:/root/.minio
#
    ports:
      - "9003:9000"
# NEW alias to be able to use wildcard cert for .local domain
    hostname: minio1.local
    networks:
       default:
         aliases:
           - minio1.local
  #
  environment:
      MINIO_ACCESS_KEY: minio
      MINIO_SECRET_KEY: minio123
# NEW variable with key's passphrase
      MINIO_CERT_PASSWD: secret
#
# CHANGED server command to https and adding .local
    command: server https://minio{1...3}.local/data{1...4}
#
    healthcheck:
# CHANGED check to use https
      test: ["CMD", "curl", "-f", "https://localhost:9000/minio/health/live"]
#
      interval: 30s
      timeout: 20s
      retries: 3

4. remove old docker volumes for minio, start minio1-3 with the --build and -d flags to docker-compose

5. encounter errors about the certs and TLS handshakes

## Context

Not able to use https yet.

## Regression

Not a regression.

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):
I am using docker-compose on a CentOS 7 VM.    
The minio containers use image: minio/minio:RELEASE.2020-08-04T23-10-51Z.

**Comment by aead** [comment]
> minio2_1 | Unable to read 'format.json' from https://minio3.local:9000/data4: Post https://minio3.local:9000/minio/storage/data4/v20/readall?disk-id=&file-path=format.json&volume=.minio.sys: x509: certificate is not valid for any names, but wanted to match minio3.local

Your certificate does not contain SANs. Note that the using the subject common name as DNS name is deprecated by the X.509 standard for decades and Go will only try to verify it if no SANs are present and the common name is an invalid host.
> Note that if the CommonName is an invalid host name, it's always ignored.

Consider using the following `openssl`commands to generate your private key and certificate:
```
openssl ecparam -genkey -name prime256v1 | openssl ec -out private.key

openssl req -new -x509 -days 3650 -key server.key -out server.cert   -subj "/C=/ST=/L=/O=/CN=" -addext "subjectAltName = DNS:*.local"
```
For more information see e.g. https://geekflare.com/san-ssl-certificate/

**Comment by MikeLReminderM** [comment]
Thank you, I will try it out.   That should probably be added to the documentation here:

https://docs.min.io/docs/how-to-secure-access-to-minio-server-with-tls

which is where I got the commands I used.

I was able to eventually get https working by creating certs for each instance and placing them all in the CAs folder for every instance, but a wildcard cert would be more efficient, especially for clusters with many instances.
